### DISCUSSION TITLE
Workaround for Selects with nullable values?

### DISCUSSION THREAD

**Discussion by PavloParafiloCricut** [Megabit/Blazorise#5056] [Ideas]
So nullable values for Select component were discussed here https://github.com/Megabit/Blazorise/issues/2921
For the code below SelectedValueChanged is not triggered when you select "ALL" option cos it has NULL value.
```
        <Select TValue="int?"  SelectedValueChanged="@((int? val) => Console.WriteLine("Changed"))">
            <SelectItem Value="(int?) null">All</SelectItem>
            <SelectItem Value="1">1</SelectItem>
            <SelectItem Value="2">2</SelectItem>
            <SelectItem Value="3">3</SelectItem>
        </Select>
```

And I see that it is something coming from Blazor itself. 
But is it possible to implement some workaround with that? 
"ALL" select list option is pretty much common and is everywhere, so we have to spread own workarounds in code.
Would be super nice to have this feature in Blazorise itself.

**Comment by David-Moreira** [comment]
As far as we know it is a limitation from html select element. Some libraries end up implementing a custom structure with div elements in order to avoid known limitations. 
However in Blazorise, we use the standard html select element.

The workaround is in the issue you've linked. Create a dummy value like -1 and turn it into null on the `SelectedValueChanged` handler.

We've also made a blog post about `Select` usage in the past if you're interested:
https://blazorise.com/blog/how-to-handle-select-with-primitive-and-complex-types
